src/request-sender.ts

```diff
@@ -1,32 +1,26 @@
-import * as isPromise from 'is-promise';
+import { CookiesStatic } from 'js-cookie';
 import { merge } from 'lodash';
+import isPromise from './is-promise';
+import PayloadTransformer from './payload-transformer';
+import RequestFactory from './request-factory';
+import { RequestOptions } from './request-options';
+import { Response } from './response';
 import Timeout from './timeout';
 
 export default class RequestSender {
-    /**
-     * @constructor
-     * @param {RequestFactory} requestFactory
-     * @param {PayloadTransformer} payloadTransformer
-     * @param {Cookie} cookie
-     */
-    constructor(requestFactory, payloadTransformer, cookie) {
-        this._requestFactory = requestFactory;
-        this._payloadTransformer = payloadTransformer;
-        this._cookie = cookie;
-    }
+    constructor(
+        private _requestFactory: RequestFactory,
+        private _payloadTransformer: PayloadTransformer,
+        private _cookie: CookiesStatic
+    ) { }
 
-    /**
-     * @param {string} url
-     * @param {RequestOptions} [options={}]
-     * @return {Promise<any>}
-     */
-    sendRequest(url, options = {}) {
+    sendRequest(url: string, options?: RequestOptions): Promise<Response> {
         const requestOptions = this._mergeDefaultOptions(options);
         const request = this._requestFactory.createRequest(url, requestOptions);
 
         return new Promise((resolve, reject) => {
             const requestHandler = () => {
-                const response = this._payloadTransformer.toResponse(request);
+                const response: Response = this._payloadTransformer.toResponse(request);
```

Comment:
I believe you can omit `Response` declaration. The compiler should be able to infer the type.